doc/source/cluster/kubernetes/examples/rayjob-kueue-gang-scheduling.md

```diff
@@ -0,0 +1,249 @@
+      - name: "nvidia.com/gpu"
+        nominalQuota: 10000 # infinite quotas
+  admissionChecks:
+  - rayjob-gpu
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: LocalQueue
+metadata:
+  namespace: "default"
+  name: "user-queue"
+spec:
+  clusterQueue: "cluster-queue"
+```
+
+Create the Kueue resources:
+```bash
+kubectl apply -f kueue-resources.yaml
+```
+
+:::{note}
+This example configures Kueue to orchestrate gang scheduling of GPUs. However, other resources such as CPU and Memory can be used.
+:::
+
+## Deploy a RayJob
+
+Download the RayJob that executes all the steps documented in [Fine-tune a PyTorch Lightning Text Classifier](https://docs.ray.io/en/master/train/examples/lightning/lightning_cola_advanced.html). The [source code](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/samples/pytorch-text-classifier) is also in the KubeRay repository.
+
+```bash
+curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/pytorch-text-classifier/ray-job.pytorch-distributed-training.yaml
+```
+
+Before creating the RayJob, modify the RayJob metadata with a label
+to assign the RayJob to the LocalQueue created earlier:
+```yaml
+metadata:
+  generateName: pytorch-text-classifier-
+    labels:
+      kueue.x-k8s.io/queue-name: user-queue
+```
+
+Deploy the RayJob:
+```bash
+$ kubectl create -f ray-job.pytorch-distributed-training.yaml
+rayjob.ray.io/dev-pytorch-text-classifier-r6d4p created
+```
+
+## Gang Scheduling with RayJob
+
+Here's the expected behavior when a GPU-requiring RayJob is deployed to a cluster initially lacking GPUs:
+* Kueue suspends the RayJob due to insufficient GPU resources in the cluster.
+* Kueue creates a ProvisioningRequest, specifying the GPU requirements for the RayJob.
+* The node autoscaler monitors ProvisioningRequests and adds nodes with GPUs as needed.
+* Once the required GPU nodes are available, the ProvisioningRequest is satisfied.
+* Kueue admits the RayJob, and its execution begins with Ray nodes scheduled on the newly provisioned nodes.
+
+In the event that GPUs are unavailable, Kueue will keep the RayJob suspended. In addition, the node autoscaler will avoid provisioning any new nodes unless it can fully satisfy the RayJob's GPU requirements.
+
+Upon creating a RayJob, notice that the RayJob is immediately suspended despite the ClusterQueue having GPU quotas available.
+```bash
+$ kubectl get rayjob pytorch-text-classifier-rj4sg -o yaml
+apiVersion: ray.io/v1
+kind: RayJob
+metadata:
+  name: pytorch-text-classifier-rj4sg
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+...
+...
+...
+status:
+  jobDeploymentStatus: Suspended  # RayJob suspended
+  jobId: pytorch-text-classifier-rj4sg-pj9hx
+  jobStatus: PENDING
+```
+
+Kueue will keep this RayJob suspended until it's corresponding ProvisioningRequest is satisfied.
+You can list ProvisioningRequest resources and their status:
+```bash
+$ kubectl get provisioningrequest
```

Comment:
However, when I increase the number of nodes in the `default-pool` from 1 to 2, it provisions a new node within 30 seconds.

Reply:
This seems to be a bug for the beta feature `gcloud beta container node-pools create gpu-node-pool ...`.

Reply:
> My provisioning request has already been created for 15 minutes, but PROVISIONED is still False.

This is expected IF there's no GPU capacity. It's possible there was no GPUs at the time you created the RayJob but then some became available when you tried to manually scale the node pool. 

> Is there any way to know the progress of the provisioning process? 

I believe there is some work planned to improve the visibility of ProvisioningRequest but I'll need to follow-up with Kueue folks on it

Reply:
> This is expected IF there's no GPU capacity. It's possible there was no GPUs at the time you created the RayJob but then some became available when you tried to manually scale the node pool.

You may have misunderstood my messages slightly. Let me summarize why I think it is a bug.

* `default-pool` is created by `gcloud container clusters create` (not a beta feature).
* `gpu-node-pool` is created by `gcloud beta container node-pools ...` (a beta feature).
* My provisioning request has already been created for 15 minutes, but PROVISIONED is still False.
* I tried to manually scale up the size of `gpu-node-pool` (https://github.com/ray-project/ray/pull/43062#discussion_r1491718392) from 0 to 1, but nothing happened (the GKE Web UI still shows the node pool has 0 node).
* I tried to manually scale up the size of `default-pool` from 1 to 2. The GKE Web UI shows the node pool has 2 nodes immediately before the new node is actually created.
* Hence, I expect that the web UI should change immediately when I edit it manually.

Reply:
> My provisioning request has already been created for 15 minutes, but PROVISIONED is still False.

PROVISIONED=False in the the ProvisioningRequest right? ProvisioningRequest does not pertain to the node pool itself, it pertains the set of nodes that will be used to run the RayJob. Provisioned=True means the nodes in the node pool have been created. 

`gpu-node-pool is created by gcloud beta container node-pools ... (a beta feature).` succeeds quickly because you create a node pool with 0 nodes.

Reply:
> Is there any way to know the progress of the provisioning process?

Regarding this, I spoke with @alculquicondor and he opened this issue as a follow-up https://github.com/kubernetes-sigs/kueue/issues/1749